File: WoofAndroid/data/repositories/src/main/kotlin/com/perrystreet/woof/repositories/dogs/mapper/DogDTOToDomainMapper.kt
package com.perrystreet.woof.repositories.dogs.mapper

import com.perrystreet.woof.dto.dog.DogDTO
import com.perrystreet.woof.models.dog.Dog
import org.koin.core.annotation.Factory

@Factory
class DogDTOToDomainMapper {
    operator fun invoke(dto: DogDTO): Dog = Dog(
        id = dto.id,
        name = dto.name,
        photoUrl = dto.photoUrl,
    )
}

File: WoofIOS/SwiftPackages/Data/Repositories/Sources/Repositories/Dogs/Mapper/DogDTOToDomainMapper.swift
import DI
import DTO
import Models

@Factory
public final class DogDTOToDomainMapper {
    public func callAsFunction(_ dto: DogDTO) -> Dog {
        Dog(
            id: dto.id,
            name: dto.name,
            photoUrl: dto.photoUrl
        )
    }
}
